.github/workflows/codeql-analysis.yml

```
@@ -14,6 +14,9 @@ on:
     #        *  * * * *
     - cron: '30 1 * * *'
 
+permissions:
+  contents: read
+
```

I think that's the minimum permission (CodeQL needs to be able to write security events, otherwise the workflow will fail)
```suggestion
permissions:
  security-events: write
  contents: read
```

---

resolved in https://github.com/open-telemetry/opentelemetry-python/pull/4663/commits/9b615a8a8830a834810dd69c394f81d49de0f7da

---

thanks for the review @pichlermarc